lib/config.js

```diff
@@ -196,13 +196,59 @@ const paramsSchema = Joi.object({
     .default()
     .unknown(false),
   clientAuthMethod: Joi.string()
-    .valid('client_secret_basic', 'client_secret_post', 'none')
+    .valid(
+      'client_secret_basic',
+      'client_secret_post',
+      'client_secret_jwt',
+      'private_key_jwt',
+      'none'
+    )
     .optional()
     .default((parent) => {
-      return parent.authorizationParams.response_type === 'id_token'
-        ? 'none'
-        : 'client_secret_basic';
-    }),
+      if (parent.authorizationParams.response_type === 'id_token') {
+        return 'none';
+      }
+      if (parent.clientAssertionSigningKey) {
+        return 'private_key_jwt';
+      }
+      return 'client_secret_basic';
+    })
+    .when(
+      Joi.ref('authorizationParams.response_type', {
+        adjust: (value) => value && value.includes('code'),
+      }),
+      {
+        is: true,
+        then: Joi.string().invalid('none').messages({
+          'any.only':
+            '"clientAuthMethod" cannot be "none" when "response_type" includes "code"',
+        }),
+      }
+    ),
+  clientAssertionSigningKey: Joi.any()
+    .optional()
+    .when(Joi.ref('clientAuthMethod'), {
+      is: 'private_key_jwt',
+      then: Joi.any().required().messages({
+        'any.required':
+          '"clientAssertionSigningKey" is required for a "clientAuthMethod" of "private_key_jwt"',
+      }),
+    }), // <Object> | <string> | <Buffer> | <KeyObject>,
+  clientAssertionSigningAlg: Joi.string()
```

Comment:
I've checked the client logic (both the v4.x we use and the latest v5.x) - if the AS discovery document does not list `token_endpoint_auth_signing_alg_values_supported` this property will turn out to be required. I think it's okay to leave like this but a note in the documentation may be worth it.